src/button/internal.tsx

```diff
@@ -221,7 +227,22 @@ export const InternalButton = React.forwardRef(
     const buttonContent = (
       <>
         <LeftIcon {...iconProps} />
-        {shouldHaveContent && <span className={clsx(styles.content, analyticsSelectors.label)}>{children}</span>}
+        {shouldHaveContent && (
+          <>
+            <span className={clsx(styles.content, analyticsSelectors.label)}>{children}</span>
+            {external && (
+              <>
+                &nbsp;
+                <span
+                  role="img"
+                  aria-label={i18n('i18nStrings.externalIconAriaLabel', i18nStrings?.externalIconAriaLabel)}
```

Comment:
Can you remind me why we did not expose `ariaLabel` prop on the Icon component yet? 

This seems like a repeating pattern already